Extensions/Spine/JsExtension.js

```diff
@@ -0,0 +1,343 @@
+    
+      object
+        .addExpressionAndConditionAndAction(
+          'number',
+          'Animation',
+          _('Animation (by number)'),
+          _(
+            'the number of the animation played by the object (the number from the animations list)'
+          ),
+          _('the number of the animation'),
+          _('Animations and images'),
+          'res/actions/animation24.png'
+        )
+        .addParameter('object', _('Spine'), 'SpineObject')
+        .useStandardParameters('number', gd.ParameterOptions.makeNewOptions())
+        .markAsSimple()
+        .setFunctionName('setAnimationIndex')
+        .setGetter('getCurrentAnimationIndex');
+
+      object
+        .addExpressionAndConditionAndAction(
+          'string',
+          'AnimationName',
+          _('Animation (by name)'),
+          _('the animation played by the object'),
+          _('the animation'),
+          _('Animations and images'),
+          'res/actions/animation24.png'
+        )
+        .addParameter('object', _('Spine'), 'SpineObject')
+        .useStandardParameters(
+          'objectAnimationName',
+          gd.ParameterOptions.makeNewOptions().setDescription(
+            _('Animation name')
+          )
+        )
+        .markAsAdvanced()
+        .setFunctionName('setAnimationName')
+        .setGetter('getAnimationName');
+
+
+    return extension;
+  },
+
+  /**
+   * You can optionally add sanity tests that will check the basic working
+   * of your extension behaviors/objects by instantiating behaviors/objects
+   * and setting the property to a given value.
+   *
+   * If you don't have any tests, you can simply return an empty array.
+   *
+   * But it is recommended to create tests for the behaviors/objects properties you created
+   * to avoid mistakes.
+   */
+  runExtensionSanityTests: function (
+    gd /*: libGDevelop */,
+    extension /*: gdPlatformExtension*/
+  ) {
+    return [];
+  },
+  /**
+   * Register editors for objects.
+   *
+   * ℹ️ Run `node import-GDJS-Runtime.js` (in newIDE/app/scripts) if you make any change.
+   */
+  registerEditorConfigurations: function (
+    objectsEditorService /*: ObjectsEditorService */
+  ) {
+    objectsEditorService.registerEditorConfiguration(
+      'SpineObject::SpineObject',
+      objectsEditorService.getDefaultObjectJsImplementationPropertiesEditor({
+        helpPagePath: '/objects/spine',
+      })
+    );
+  },
+  /**
+   * Register renderers for instance of objects on the scene editor.
+   *
+   * ℹ️ Run `node import-GDJS-Runtime.js` (in newIDE/app/scripts) if you make any change.
+   */
+  registerInstanceRenderers: function (
+    objectsRenderingService /*: ObjectsRenderingService */
+  ) {
+    const { PIXI, RenderedInstance, gd } = objectsRenderingService;
+
+    class RenderedSpineInstance extends RenderedInstance {
+      _spine;
+      _rect = new PIXI.Graphics();
+      _initialWidth;
+      _initialHeight;
+      _animationIndex;
```

Comment:
What are these?

Reply:
It doesnt work for some reasons. Face exceptions them